.github/workflows/check-license.yml

```diff
@@ -0,0 +1,60 @@
+name: Validate SPDX License year and company mention in currently changed Files
+
+on:
+  pull_request:
+    branches: [main]
+    types:
+      - opened
+      - reopened
+      - synchronize
+      - assigned
+      - review_requested
+
+jobs:
+  check-spdx-license:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 2
+
+      - name: Identify currently changed files and check licenses
+        id: check-licenses
+        run: |
+          current_year=$(date +"%Y")  # Dynamically set current year for checking
+          echo "current_year=$current_year" >> $GITHUB_ENV  # Export current year to environment variables
+          missing_license_files=""
+          for file in $(git diff --name-only HEAD^ HEAD | grep '\.py$'); do
+            # Check if file contains the correct SPDX license identifier and the exact copyright statement with dynamic current year
+            if ! grep -q "SPDX-License-Identifier: Apache-2.0" "$file" || ! grep -q "SPDX-FileCopyrightText: © $current_year Tenstorrent AI ULC" "$file"; then
+              missing_license_files="$missing_license_files$file "
+            fi
+          done
+          echo "missing_license_files=$missing_license_files" >> $GITHUB_ENV
+          echo "Checked files for correct SPDX-License compliance."
+
+      - name: Comment on PR about incorrect SPDX licenses
+        if: env.missing_license_files != ''
+        uses: actions/github-script@v6
+        with:
+          script: |
+            const issueNumber = context.issue.number;
+            const owner = context.repo.owner;
+            const repo = context.repo.repo;
+            const missingFiles = process.env.missing_license_files.trim().replace(/\s+/g, '\n');
+            const currentYear = process.env.current_year;
+            const commentBody = `The following Python files are not compliant with the required licensing standards for ${currentYear} or do not contain the correct copyright text 'Tenstorrent AI ULC' within their license header:\n\`\`\`\n${missingFiles}\n\`\`\`\nPlease update the files accordingly. Your attention and cooperation in this matter are greatly appreciated.\nThank you.`;
```

Comment:
Should we remove "Python files" from here and just say "files"?

Would this work with other file formats i.e. .c, .cpp, etc.?

I think we would just need to add the extension to `check_copyright_config.yaml` but then it should check all of the formats?

Reply:
> Should we remove "Python files" from here and just say "files"?

I just updated it to just say files ; That's a good catch. 

> Would this work with other file formats i.e. .c, .cpp, etc.?

Currently i dont think so , `license-checker.yml` uses the `check_copyright_config.yaml` which based on [this ](https://github.com/anirudTT/benchmarking/blob/main/check_copyright_config.yaml#L8) should check on other files but frankly I need to test this; as I have only been testing python files.

> I think we would just need to add the extension to check_copyright_config.yaml but then it should check all of the formats?

Yes ; to check other files ; but I think currently let's keep it this way and based on whats gets added update the config , what do you think?

Reply:
Exactly -- no action; this was more out of my curiosity. When a different file format comes in we will test that.

Reply:
I think let me also test this on my dev branch -- just to figure out how its going to act